compiler/src/iree/compiler/Codegen/Common/DecomposeAffineOpsPass.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "mlir/Dialect/Affine/Transforms/Transforms.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+
+struct DecomposeAffineOpsPass
+    : public DecomposeAffineOpsBase<DecomposeAffineOpsPass> {
+  void runOnOperation() override;
+};
+
+}  // namespace
+
+void DecomposeAffineOpsPass::runOnOperation() {
+  IRRewriter rewriter(&getContext());
+  this->getOperation()->walk([&](AffineApplyOp op) {
+    rewriter.setInsertionPoint(op);
+    reorderOperandsByHoistability(rewriter, op);
+    (void)decompose(rewriter, op);
```

Comment:
nit: should we check the result and signal pass failure if it fails?
```
      if (failed(decompose(rewriter, op)) {
        return signalPassFailure();
      }
```

Reply:
I don't think we want to signal a failure when this "fails".
The failures happen when we encounter `affine.apply`s that cannot be rewritten (https://github.com/llvm/llvm-project/blob/fa326c21919faea33b0470b7083dccbd3c1fc3f2/mlir/include/mlir/Dialect/Affine/Transforms/Transforms.h#L37) and for these I think it is okay to just skip them.

Reply:
Its OK for doing that now, but that might mean the `decompose` method needs to return state better. If it is not a real failure that needs to be caught by the compiler, it shouldnt return failure. Would be good to fix (but I dont care enough to make it required)

Reply:
@nicolasvasilache what do you think?

I’m thinking that maybe we want to return a simple bool that would report if anything was changed.